include/lbann/layers/activations/relu.hpp

```diff
@@ -75,6 +75,11 @@ class relu_layer : public data_type_layer<TensorDataType>
   std::string get_type() const override { return "ReLU"; }
   data_layout get_data_layout() const override { return T_layout; }
   El::Device get_device_allocation() const override { return Dev; }
+  bool can_run_inplace() const override { return true; }
+  int get_backprop_requirements() const override
+  {
+    return ERROR_SIGNALS | ACTIVATIONS;
```

Comment:
@tbennun Am I missing something, but I think that this should only rely on `PREV_ACTIVATIONS`

Reply:
```suggestion
    return ERROR_SIGNALS | PREV_ACTIVATIONS;
```

Reply:
ReLU is a special case: if the output activations are zero, then the input error signals should be zero as well. Otherwise everything is kept equal. This is the property that we use to run the layer in-place.